flocker/node/agents/test/test_blockdevice.py

```diff
@@ -894,6 +894,42 @@ def test_no_delete_if_in_use(self):
             in_parallel(changes=[]),
         )
 
+    def test_deleted_dataset_volume_unmounted(self):
+        """
+        ``DestroyBlockDeviceDataset`` is a compound state change that first
+        attempts to unmount the block device.
+        Therefore do not calculate deletion for blockdevices that are not
+        manifest.
+        """
+        local_state = self.ONE_DATASET_STATE
+        local_config = to_node(local_state).transform(
+            ["manifestations", unicode(self.DATASET_ID), "dataset", "deleted"],
+            True
+        )
+        # Remove the manifestation and its mount path.
+        local_state = local_state.transform(
+            ['manifestations', unicode(self.DATASET_ID)],
+            discard
+        )
+        local_state = local_state.transform(
+            ['paths', unicode(self.DATASET_ID)],
+            discard
+        )
+
+        assert_calculated_changes(
+            self, local_state, local_config, set(),
+            in_parallel(
+                changes=[
+                    MountBlockDevice(
+                        mountpoint=FilePath('/flocker/').child(
+                            unicode(self.DATASET_ID)
+                        ),
+                        dataset_id=self.DATASET_ID
+                    )
+                ]
+            ),
```

Comment:
I know what you mean but as it stands, the list of creates is generated from a filtered list of configured datasets where `deleted=False`.
Only then does it check the cluster state for existing manifest and non-manifest datasets.

A Mount rather than an Attach is calculated because we're supplying state where there's a device file but not a mount path for a dataset.